I pulled this branch and saw the failure after about 5-6 minutes.

```
--- FAIL: TestTCPCertExpiration (22.42s)
    appaccess_test.go:661:
        	Error Trace:	/Users/zmb/teleport/integration/appaccess/appaccess_test.go:661
        	Error:      	timeout waiting for monitor channel signal
        	Test:       	TestTCPCertExpiration
```

--- a/integration/appaccess/appaccess_test.go
+++ b/integration/appaccess/appaccess_test.go
@@ -669,12 +669,16 @@ func TestTCPCertExpiration(t *testing.T) {
 	// Close and re-open the connection
 	require.NoError(t, conn.Close())
 
-	conn, err = net.Dial("tcp", address)
-	require.NoError(t, err)
+	require.Eventually(t, func() bool {
+		conn, err := net.Dial("tcp", address)
+		if err != nil {
+			return false
+		}
 
-	// Try to read again, expect a failure.
-	_, err = conn.Read(buf)
-	require.Error(t, err, buf)
+		// Try to read again, expect a failure.
+		_, err = conn.Read(buf)
+		return err != nil
+	}, time.Second*5, time.Millisecond*100)
 }
 
 func testServersHA(p *Pack, t *testing.T) {

--- a/integration/appaccess/appaccess_test.go
+++ b/integration/appaccess/appaccess_test.go
@@ -571,17 +571,28 @@ func TestTCPLock(t *testing.T) {
 
 	// Start the proxy to the two way communication app.
 	address := pack.startLocalProxy(t, pack.rootTCPTwoWayPublicAddr, pack.rootAppClusterName)
-	conn, err := net.Dial("tcp", address)
-	require.NoError(t, err)
 
-	// Read, write, and read again to make sure its working as expected.
+	var conn net.Conn
+	var err error
+	var n int
 	buf := make([]byte, 1024)
-	n, err := conn.Read(buf)
-	require.NoError(t, err, buf)
+
+	// Try to read for a short amount of time.
+	require.Eventually(t, func() bool {
+		conn, err = net.Dial("tcp", address)
+		if err != nil {
+			return false
+		}
+
+		// Try to read
+		n, err = conn.Read(buf)
+		return err == nil
+	}, time.Second*5, time.Millisecond*100)
 
 	resp := strings.TrimSpace(string(buf[:n]))
 	require.Equal(t, pack.rootTCPTwoWayMessage, resp)
 
+	// Once we've read successfully, write, and then read again to verify the connection
 	_, err = conn.Write(msg)
 	require.NoError(t, err)
 
@@ -621,7 +632,7 @@ func TestTCPLock(t *testing.T) {
 // TestTCPCertExpiration tests TCP application with certs expiring.
 func TestTCPCertExpiration(t *testing.T) {
 	clock := clockwork.NewFakeClockAt(time.Now())
-	mCloseChannel := make(chan struct{})
+	mCloseChannel := make(chan struct{}, 1)
 	pack := SetupWithOptions(t, AppTestOptions{
 		Clock:               clock,
 		MonitorCloseChannel: mCloseChannel,
@@ -631,17 +642,28 @@ func TestTCPCertExpiration(t *testing.T) {
 
 	// Start the proxy to the two way communication app.
 	address := pack.startLocalProxy(t, pack.rootTCPTwoWayPublicAddr, pack.rootAppClusterName)
-	conn, err := net.Dial("tcp", address)
-	require.NoError(t, err)
 
-	// Read, write, and read again to make sure its working as expected.
+	var conn net.Conn
+	var err error
+	var n int
 	buf := make([]byte, 1024)
-	n, err := conn.Read(buf)
-	require.NoError(t, err, buf)
+
+	// Try to read for a short amount of time.
+	require.Eventually(t, func() bool {
+		conn, err = net.Dial("tcp", address)
+		if err != nil {
+			return false
+		}
+
+		// Read, write, and read again to make sure its working as expected.
+		n, err = conn.Read(buf)
+		return err == nil
+	}, time.Second*5, time.Millisecond*100)
 
 	resp := strings.TrimSpace(string(buf[:n]))
 	require.Equal(t, pack.rootTCPTwoWayMessage, resp)
 
+	// Once we've read successfully, write, and then read again to verify the connection
 	_, err = conn.Write(msg)
 	require.NoError(t, err)
 

--- a/integration/appaccess/appaccess_test.go
+++ b/integration/appaccess/appaccess_test.go
@@ -561,7 +561,7 @@ func TestTCP(t *testing.T) {
 // TestTCPLock tests locking TCP applications.
 func TestTCPLock(t *testing.T) {
 	clock := clockwork.NewFakeClockAt(time.Now())
-	mCloseChannel := make(chan struct{})
+	mCloseChannel := make(chan struct{}, 1)
 	pack := SetupWithOptions(t, AppTestOptions{
 		Clock:               clock,
 		MonitorCloseChannel: mCloseChannel,

